src/org/labkey/test/pages/core/admin/AllowedFileExtensionAdminPage.java

```diff
@@ -0,0 +1,189 @@
+        return new AllowedFileExtensionAdminPage(webDriverWrapper.getDriver());
+    }
+
+    @Override
+    protected void waitForPage()
+    {
+        waitFor(()-> {
+            try
+            {
+                return !BootstrapLocators.loadingSpinner.areAnyVisible(getDriver()) &&
+                        elementCache().extension.getComponentElement().isDisplayed();
+            }
+            catch (NoSuchElementException | StaleElementReferenceException | TimeoutException retry)
+            {
+                return false;
+            }
+        }, "Allowed File Extensions page did not load in time.", 3_000);
+
+    }
+
+    public AllowedFileExtensionAdminPage setExtension(String extension)
+    {
+        elementCache().extension.set(extension);
+        return this;
+    }
+
+    public String clickSaveExpectingError()
+    {
+        String errorText;
+        elementCache().saveExtension.click();
+        errorText = waitForElement(Locators.labkeyError).getText();
+        clearCache();
+        return errorText;
+    }
+
+    public AllowedFileExtensionAdminPage clickSaveExtension()
+    {
+        elementCache().saveExtension.click();
+        clearCache();
+        return this;
+    }
+
+    public AllowedFileExtensionAdminPage updateExtension(String oldExtension, String newExtension)
+    {
+        getAllowedExtension(getAllowedExtensionIndex(oldExtension)).set(newExtension);
+        clearCache();
+        return this;
+    }
+
+    public AllowedFileExtensionAdminPage clickSaveUpdateExtension()
+    {
+        elementCache().saveUpdateExtension.click();
+        sleep(750);
+        clearCache();
+        return this;
+    }
+
+    public String clickUpdateExtensionExpectingError()
+    {
+        elementCache().saveUpdateExtension.click();
+        clearCache();
+        return waitForElement(Locators.labkeyError).getText();
+    }
+
+    public AllowedFileExtensionAdminPage deleteExtension(int index)
+    {
+        WebElement deleteButton = elementCache().deleteExtensions.get(index);
+        deleteButton.click();
+
+        shortWait().withMessage("Existing extenstion was not deleted.")
+                .until(ExpectedConditions.stalenessOf(deleteButton));
+
+        clearCache();
+        return this;
+    }
+
+    public AllowedFileExtensionAdminPage deleteExtension(String extension)
+    {
+        return deleteExtension(getAllowedExtensionIndex(extension));
+    }
+
+    public AllowedFileExtensionAdminPage deleteAllExtensions(boolean acceptAlert)
+    {
+        elementCache().deleteAll.click();
+
+        if (acceptAlert)
+        {
+            acceptAlert();
+            shortWait().withMessage("'Delete All' button should have gone away.")
+                    .until(ExpectedConditions.stalenessOf(elementCache().deleteAll));
+        }
+        else
+        {
+            cancelAlert();
+        }
+
+        clearCache();
+        return this;
```

Comment:
This staleness check will always pass because there is a page load. Need to clear the cache and check the button's visibility.
```suggestion
        if (acceptAlert)
        {
            doAndWaitForPageLoad(() -> {
                elementCache().deleteAll.click();
                acceptAlert();
            });
            clearCache();
            assertFalse("Delete All button should not be present after deleting all extensions", elementCache().deleteAll.isDisplayed());
        }
        else
        {
            elementCache().deleteAll.click();
            cancelAlert();
            assertTrue("Delete All button should be present", elementCache().deleteAll.isDisplayed());
        }

        return this;
```

Reply:
Updated.